src/librustdoc/html/templates/type_layout.html

```diff
@@ -0,0 +1,45 @@
+<h2 id="layout" class="small-section-header">  {# #}
+    Layout<a href="#layout" class="anchor">§</a> {# #}
+</h2> {# #}
+<div class="docblock"> {# #}
+    {% match self.type_layout() %}
+        {% when Ok(ty_layout) %}
+            <div class="warning"> {# #}
+                <p> {# #}
+                    <strong>Note:</strong> Most layout information is <strong>completely {#+ #}
+                    unstable</strong> and may even differ between compilations. {#+ #}
+                    The only exception is types with certain <code>repr(...)</code> {#+ #}
+                    attributes. Please see the Rust Reference’s {#+ #}
+                    <a href="https://doc.rust-lang.org/reference/type-layout.html">“Type Layout”</a> {#+ #}
+                    chapter for details on type layout guarantees. {# #}
+                </p> {# #}
+            </div> {# #}
+            <p><strong>Size:</strong> {{ self.write_size(ty_layout.layout.0.borrow(), 0) | safe }}</p> {# #}
+            {% if let Some(variants) = self.variants() %}
```

Comment:
In templates, for repeated elements where an empty list is equivalent to None, I think I prefer

    {% if !self.variants.is_empty() %}
        <p><strong>Size for each variant:</strong></p>
        ... do something for each variant

I realize using Options is a little more Rusty in general but for templates in particular the fundamental concepts are conditionals and repetition, and the more we can stick to those basics the easier they will be to understand. Also I think anytime we're doing variable bindings (`if let`) inside a template risks developing into too much complexity.